File: Algorithm/SWEA/D3_1213.cpp
//https://www.swexpertacademy.com/main/code/problem/problemDetail.do?contestProbId=AV14P0c6AAUCFAYi&categoryId=AV14P0c6AAUCFAYi&categoryType=CODE
//http://1ilsang.blog.me/221323675039

#include <stdio.h>
#include <iostream>
#include <string>
using namespace std;

int main(void){
    for(int tcase = 1; tcase <= 10; tcase++){
        int tmp, ans = 0;
        scanf("%d", &tmp);
        string a;
        string b;
        cin >> a;
        cin >> b;
        for(int i = 0; i < b.size(); i++){
                int k = i, cnt = 0;
            for(int j = 0; j < a.size(); j++){
                if(b[k] == a[j]){
                    k++;
                    cnt++;
                    //printf(" %c %d %c %d\n", b[k],k, a[j], cnt);
                    if(cnt == a.size()){
                        ans++;
                    }
                }
                else {
                    break;}
            }
        }
        printf("#%d %d\n", tcase, ans);
    }
    return 0;
}

File: Algorithm/SWEA/java/D3_1213.java
//https://www.swexpertacademy.com/main/code/problem/problemDetail.do?contestProbId=AV14P0c6AAUCFAYi

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        for(int t = 1; t < 11; t++) {
            int dump = Integer.parseInt(br.readLine());
            String target = br.readLine();
            String str = br.readLine();
            int ans = 0;
            for(int i = 0; i < str.length(); i++) {
                boolean flag = true;
                if(i == str.length() - 1 && target.length() > 0) continue;
                for(int j = 0 ; j < target.length(); j++) {
                    if(str.charAt(i + j) != target.charAt(j)) {
                        flag = false;
                        break;
                    }
                }
                if(flag) ans++;
            }
            System.out.printf("#%d %d\n", t, ans);
        }
        br.close();
    }
}
